runtime/parachains/src/inclusion.rs

```diff
@@ -692,6 +713,34 @@ const fn availability_threshold(n_validators: usize) -> usize {
 	threshold
 }
 
+#[derive(derive_more::From, Debug)]
+enum AcceptanceCheckErr<BlockNumber> {
+	HeadDataTooLarge,
+	PrematureCodeUpgrade,
+	NewCodeTooLarge,
+	ProcessedDownwardMessages(router::ProcessedDownwardMessagesAcceptanceErr),
+	UpwardMessages(router::UpwardMessagesAcceptanceCheckErr),
+	HrmpWatermark(router::HrmpWatermarkAcceptanceErr<BlockNumber>),
+	OutboundHrmp(router::OutboundHrmpAcceptanceErr),
+}
+
+impl<BlockNumber> AcceptanceCheckErr<BlockNumber> {
+	/// Returns the same error so that it can be threaded through a needle of `DispatchError` and
+	/// ultimately returned from a `Dispatchable`.
+	fn strip_into_dispatch_err<T: Trait>(self) -> Error<T> {
```

Comment:
I deliberately went with a inherent method rather than a `From` impl. This way we have an opportunity to leave a comment and IMO `. strip_into_dispatch_err` gives a little bit better sense of what would happen compared to a `.into()`.

Reply:
You can also put a comment above a `From` implementation :shrug: 

Plus, you will need to do `Error::<T>::from` anyway.

Reply:
True, but I still prefer a specific `.strip_into_dispatch_err()` to generic `.into` (or `from`) at the call site and the comment would show up in the rustdoc (if you are using a proper IDE anyway lol).

Reply:
I mean you would need to write:
```
Error::<T>::from(err)
```

Just calling `err.into()` would not work.

Reply:
Ah, yeah, gotcha.